datafusion/sql/src/planner.rs

```diff
@@ -1613,6 +1613,13 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                                 &schema,
                                 &mut HashMap::new(),
                             ),
+                        SQLExpr::TypedString {
+                            ref data_type,
+                            ref value,
+                        } => Ok(Expr::Cast {
+                            expr: Box::new(lit(&**value)),
+                            data_type: convert_data_type(data_type)?,
+                        }),
```

Comment:
In case you wanted to make this work with somewhat fewer `*` and `&` you can do something like:

```suggestion
                        SQLExpr::TypedString {
                            data_type,
                            value,
                        } => Ok(Expr::Cast {
                            expr: Box::new(lit(value)),
                            data_type: convert_data_type(&data_type)?,
                        }),
```

Reply:
Cool – I'll make that change as it is much cleaner! I copied this pattern from another part of the file:

https://github.com/apache/arrow-datafusion/blob/c46d61d45eb73f11ac39d8646b2b25a4fbbc80eb/datafusion/sql/src/planner.rs#L1818-L1824

so I've updated that too.